### ISSUE TITLE
HTTP Connection not Released (SingleClientConnManager)

### ISSUE THREAD

**Issue by nix-codes** [germanklf/maven-m2e-codestyle#1]
Hi

I have just tried this plugin. It looks like there's a problem with the HTTP connections. When this plugin runs the second time, I get the following error:

```
Execution default of goal com.despegar.maven.plugin:maven-m2e-codestyle:1.0.3:configure failed: Invalid use of SingleClientConnManager: connection still allocated.
Make sure to release the connection before allocating another one. (com.despegar.maven.plugin:maven-m2e-codestyle:1.0.3:configure:default:process-resources)
```

This is how I configured the m2e lifecycle mapping in the 'pluginManagement' section of my POM:

``` xml
<plugin>
  <groupId>org.eclipse.m2e</groupId>
  <artifactId>lifecycle-mapping</artifactId>
  <version>1.0.0</version>
  <configuration>
    <lifecycleMappingMetadata>
      <pluginExecutions>
        <pluginExecution>
          <pluginExecutionFilter>
            <groupId>com.despegar.maven.plugin</groupId>
            <artifactId>maven-m2e-codestyle</artifactId>
            <versionRange>[1.0.3,)</versionRange>
            <goals>
              <goal>configure</goal>
            </goals>
          </pluginExecutionFilter>
          <action>
            <execute />
          </action>
        </pluginExecution>
      </pluginExecutions>
    </lifecycleMappingMetadata>
  </configuration>
</plugin>
```

I'm using Eclipse Juno (4.2) with m2e 1.2.0.20120903-1050

**Comment by germanklf** [comment]
I think that Eclipse is running a single instance of the Mojo, maybe this is a change for Juno, that is firing this processes concurrently.

I'll try changing this line of "MavenM2ECodeStyleMojo":
`private HttpClient httpClient = new DefaultHttpClient();`

To:
`private HttpClient httpClient = new DefaultHttpClient(new ThreadSafeClientConnManager());`

**Comment by nix-codes** [comment]
Sure. Let me know if you build a snapshot and I can test it. Thanks.

**Comment by germanklf** [comment]
I've just uploaded 1.0.4-SNAPSHOT to the OSS repo (https://oss.sonatype.org/content/repositories/snapshots/).

In my Juno installation worked fine, let me know if you find any issue.

**Comment by germanklf** [comment]
I released 1.0.4, it will be available soon on the OSS repo and maven central.
